packages/core-utils/src/route.js

```diff
@@ -1,31 +1,151 @@
 /**
- * Looks up an operator from the provided configuration.
- * @param  {string} feedId - the feed ID, which can be obtained by splitting the
- *                            OTP route or stop ID by `:`
- * @param  {object} transitOperators - transitOperators from config.
- * @return {object} the operator for the specified feedId
+ * Returns the transit operator (if an exact match is found) from the transit
+ * operators config value. It is critical to use both the feedId and agencyId in
+ * this method because it is possible in OTP for there to be a duplicate
+ * agencyId in separate feeds.
+ *
+ * @param  {string} feedId The feedId that this transit agency belongs to
+ * @param  {string} agencyId The agencyId of the transit agency
+ * @param  {array} transitOperators The transitOperators list from the config
+ * @return {object} The transitOperator if a match was found or null if no match
+ *    was found
  */
-export function getTransitOperatorFromId(feedId, transitOperators) {
+export function getTransitOperatorFromFeedIdAndAgencyId(
+  feedId,
+  agencyId,
+  transitOperators
+) {
   return (
-    transitOperators.find(transitOperator => transitOperator.id === feedId) ||
-    null
+    transitOperators.find(
+      transitOperator =>
+        transitOperator.feedId === feedId &&
+        transitOperator.agencyId === agencyId
+    ) || null
   );
 }
 
-/** Looks up an operator from the provided configuration */
+/**
+ * Looks up an operator from the provided leg.
+ *
+ * @param  {object} leg The Itinerary Leg from which to find the transit
+ *    operator
+ * @param  {object} transitOperators transitOperators from config.
+ * @return {object} the operator if one was found or null if no match was found
+ */
+export function getTransitOperatorFromLeg(leg, transitOperators) {
+  if (!leg.routeId || !leg.agencyId) return null;
+  const feedId = leg.routeId.split(":")[0];
+  return getTransitOperatorFromFeedIdAndAgencyId(
+    feedId,
+    leg.agencyId,
+    transitOperators
+  );
+}
+
+/**
+ * Looks up an operator from the provided configuration given an OTP route.
+ * NOTE: this assumes the use of the OTP Route model or a modified OTP
+ * RouteShort model (such as the one found in the IBI fork of OTP) that also
+ * returns the agencyId.
```

Comment:
Is there a PR to the main OTP repo to add the agencyId field? If not, we should prioritize that.

Reply:
It is included in https://github.com/ibi-group/OpenTripPlanner/pull/49. See https://github.com/ibi-group/OpenTripPlanner/pull/49/files#diff-139cb730f647135718a77f533420f7c5d731ac996a9e32c8bae292184c68ec87.